sympy/logic/tests/test_lra_theory.py

```diff
@@ -0,0 +1,505 @@
+from sympy.core.numbers import Rational, I, oo
+from sympy.core.relational import Eq
+from sympy.core.symbol import symbols
+from sympy.core.singleton import S
+from sympy.matrices.dense import Matrix
+from sympy.matrices.dense import randMatrix
+from sympy.assumptions.ask import Q
+from sympy.logic.boolalg import And
+from sympy.abc import x, y, z
+from sympy.assumptions.cnf import CNF, EncodedCNF
+from sympy.functions.elementary.trigonometric import cos
+from sympy.external import import_module
+
+from sympy.logic.algorithms.lra_theory import LRASolver, UnhandledInput, LRARational, HANDLE_NEGATION
+from sympy.core.random import random, choice, randint
+from sympy.core.sympify import sympify
+from sympy.ntheory.generate import randprime
+
+from sympy.testing.pytest import raises, XFAIL, skip
+import time
+
+def make_random_problem(num_variables=2, num_constraints=2, sparsity=.1, rational=True,
```

Comment:
This doesn't need to be done in this pull request, but I'd like to investigate replacing  this (or at least adding in addition to this) with hypothesis strategies.